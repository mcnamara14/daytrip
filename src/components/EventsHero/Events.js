import React from 'react';
import './EventsHero.css';

export const EventsHero = () =>  {
  return (
    <div className="eventsHeroContainer">     
      <article className="eventsHeroTextBox">
        <h1>Rockies vs Reds</h1>
        <h4><span>Coors Field</span></h4>
        <button>Select Event</button>
      </article>
    </div>
  );
};

export default EventsHero;
