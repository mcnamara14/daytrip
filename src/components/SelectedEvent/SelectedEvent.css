.selectedEvent {
  height: 300px;
  display: block;
  background: red;
}
