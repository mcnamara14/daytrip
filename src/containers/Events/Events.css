* {
  font-family: "Work Sans";
}

h1 {
  font-family: "Exo 2";
  font-size: 2rem;
  color: #fff;
  text-transform: uppercase;
  font-style: italic;
  font-weight: 500;
}

h2 {
  color: #787878;
  font-family: "Exo 2";
  font-size: 1.2rem;
  margin: 0;
}

h3 {
  color: #787878;
  font-family: "Exo 2";
  font-weight: 500;
  text-align: left;
  font-size: 1.1rem;
  margin-top: 0px;
}

.loginContainer {
  height: 685px;
  padding-top: 140px;
  background-image: url('./assets/login-bg.jpg');
  background-size: cover;
}

.signupContainer {
  width: 570px;
  margin: 50px auto 0 auto;
  background: #fff;
  border-radius: 5px;
  padding: 40px 70px;
}
