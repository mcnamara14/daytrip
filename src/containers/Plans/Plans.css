.plansContainer {
  width: 86%;
  margin: 0 auto;
  padding-top: 60px;
  display: flex;
  flex-direction: column;
}

.restaurantImage {
  width: 90%;
  height: 190px;
  border-radius: 5px 5px 0 0;
  display: block;
  border-radius: 5px;
  background-size: cover;
  background-position: center;
  border: 1px solid #eee;
}

.plansContainer h3 {
  font-size: 1.5rem;
  color: #494949;
  text-transform: uppercase;
  text-align: left;
}

.plansContainer h3 {
color: #7c7c7c;
font-size: 1.2rem;
text-transform: uppercase;
}