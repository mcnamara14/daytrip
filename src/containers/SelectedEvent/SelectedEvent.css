.selectedEvent {
  height: 300px;
  background: #282829;
  display: flex;
  background: url('./assets/selected-bg.jpg');
  background-size: cover;
  background-position: center;
}

.selectedImage {
  width: 340px;
  height: 225px;
  background-size: cover;
  border-radius: 3px;
  background-position: center;
}

.selectedContainer {
  width: 1200px;
  margin: 0 auto;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.selectedTitle {
  width: 33%;
  text-align: left;
}

.selectedTitle h3 {
  font-family: "Roboto";
  text-transform: uppercase;
  font-size: 2.2rem;
  margin-bottom: 20px;
}

.selectedTitle p {
  color: #c2c2c2;
  text-transform: uppercase;
}

.selectedInfo {
  width: 30%;
  text-align: left;
  display: flex;
  align-items: center;
}

.selectedPrice {
  font-size: 3rem;
  color: #fff;
  font-family: "Roboto";
  margin-right: 20px;
}

.selectedVenue {
  color: #fff;
}

.selectedDate {
  color: #c2c2c2;
}