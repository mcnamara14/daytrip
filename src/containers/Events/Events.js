import React, { Component } from 'react';
import { connect } from 'react-redux';
import * as authorization from '../../firebase/auth';
import './Events.css';
import { RecentEvent } from '../../components/RecentEvent/RecentEvent';

export class Events extends Component {

  render() {
    const recentEvents = this.props.events.map((event, index) => {
      return (
        <RecentEvent
          key={index}
          image={event.image}
          title={event.title}
          price={event.price}
          venue={event.venue}
          date={event.date}
        />
      )
    })
    console.log(recentEvents)
    return (
      <div className="eventsContainer">
        <h1>Sign up to get started</h1>
        <section className="recentEvents">
          { recentEvents }
        </section>
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  events: state.recentEvents
})

export default connect(mapStateToProps)(Events);
