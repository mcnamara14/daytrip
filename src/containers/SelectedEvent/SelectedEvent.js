import React, { Component } from 'react';
import { connect } from 'react-redux';
import './SelectedEvent.css';
import PropTypes from 'prop-types';

export class SelectedEvent extends Component {
  render() {
    const { image, title, price, venue, date } = this.props.selectedEvent;
    const backgroundImage = {backgroundImage: "url(" + image + ")"};
  
    return (
      <div className="selectedEvent">
        <div className="selectedContainer">
          <div className="selectedImage" style={ backgroundImage } ></div>
          <div className="selectedTitle">
            <h3>{title}</h3>
            <p>Your selected event</p>
          </div>
          <div className="selectedInfo">
            <p className="selectedPrice">{price}</p>
            <span>
              <p className="selectedVenue">{venue}</p>
              <p className="selectedDate">{date}</p>
            </span>
          </div>
        </div>
      </div>
    );
  } 
}

SelectedEvent.propTypes = {
  selectedEvent: PropTypes.object
};

export const mapStateToProps = (state) => ({
  selectedEvent: state.selectedEvent
});

export default connect(mapStateToProps)(SelectedEvent);
