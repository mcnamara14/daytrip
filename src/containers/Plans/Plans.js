import React, { Component } from 'react';
import { connect } from 'react-redux';
import './Plans.css';
import Stop from '../Stop/Stop';
import PropTypes from 'prop-types';
import * as firebase from 'firebase';
import 'firebase/database';
import { withRouter } from 'react-router-dom';

export class Plans extends Component {
  selectPlan = (
    restaurantId,
    restaurantName, 
    restaurantRating, 
    restaurantAddress, 
    restaurantPrice, 
    restaurantReviewCount, 
    restaurantImage,
    barId,
    barName, 
    barRating, 
    barAddress, 
    barPrice, 
    barReviewCount, 
    barImage,
    id,
    title, 
    image, 
    venue, 
    date,
    eventPrice,
    planNum,
    url
  ) => {
    let firebaseLocation;
    const firebaseRef = firebase.database().ref();

    this.props.user.userId !== null ? firebaseLocation = firebaseRef.child('users').child(this.props.user.userId).child('selectedPlan') :
      firebaseLocation = firebaseRef.child('users').child('anonymous').child('selectedPlan');

    firebaseLocation.update(
      {
        restaurant: {
          id: restaurantId,
          title: restaurantName,
          rating: restaurantRating,
          location: restaurantAddress, 
          price: restaurantPrice, 
          reviews: restaurantReviewCount, 
          image: restaurantImage, 
          type: 'restaurant', 
          index: planNum
        },
        bar: {
          id: barId,
          title: barName,
          rating: barRating,
          location: barAddress, 
          price: barPrice,
          reviews: barReviewCount,
          image: barImage,
          type: 'bar' ,
          index: planNum
        },
        event: {
          id: id, 
          title: title, 
          rating: 0, 
          location: venue, 
          price: eventPrice, 
          reviews: date, 
          image: image, 
          type: 'event', 
          index: planNum,
          tickets: url
        }
      }
    );

    this.props.history.push('/plan');
  }

  getPlans = () => {
    const { 
      suggestedBars, 
      suggestedRestaurants, 
      selectedEvent 
    } = this.props;
    let plans = [];

    if (suggestedBars.length && suggestedRestaurants.length) {
      for (var i = 0; i < this.props.suggestedBars.length; i++) {
        const { 
          id: restaurantId,
          name: restaurantName, 
          rating: restaurantRating, 
          address: restaurantAddress, 
          price: restaurantPrice, 
          review_count: restaurantReviewCount, 
          image_url: restaurantImage 
        } = this.props.suggestedRestaurants[i];
        const { 
          id: barId,
          name: barName, 
          rating: barRating, 
          address: barAddress, 
          price: barPrice, 
          review_count: barReviewCount, 
          image_url: barImage 
        } = this.props.suggestedBars[i];
        const { 
          id,
          title, 
          image, 
          venue, 
          date,
          url } = this.props.selectedEvent;
        const eventPrice = selectedEvent.price;
        const planNum = i + 1;

        plans.push(<div key={i} className="planContainer">
          <h4>Plan {i + 1}</h4>
          <section className="plan">
            <Stop 
              id={restaurantId}
              title={restaurantName} 
              rating={restaurantRating} 
              location={restaurantAddress} 
              price={restaurantPrice} 
              reviews={restaurantReviewCount} 
              image={restaurantImage} 
              type={'restaurant'} 
              index={planNum}
            />
            <img src={require('./assets/route-img.jpg')} className="routeImg" />
            <Stop
              id={id} 
              title={title} 
              rating={0} 
              location={venue} 
              price={eventPrice} 
              reviews={date} 
              image={image} 
              type={'event'} 
              index={planNum}
            />
            <img src={require('./assets/route-img.jpg')} className="routeImg" />
            <Stop 
              id={barId}
              title={barName} 
              rating={barRating} 
              location={barAddress} 
              price={barPrice} 
              reviews={barReviewCount} 
              image={barImage} 
              type={'bar'} 
              index={planNum}
            />
          </section>
          <button onClick={() => this.selectPlan(
            restaurantId,
            restaurantName, 
            restaurantRating, 
            restaurantAddress, 
            restaurantPrice, 
            restaurantReviewCount, 
            restaurantImage,
            barId,
            barName, 
            barRating, 
            barAddress, 
            barPrice, 
            barReviewCount, 
            barImage,
            id,
            title, 
            image, 
            venue, 
            date,
            eventPrice,
            planNum,
            url
          )}>Select Plan</button>
        </div>);
      }
    }

    return plans;
  }

  render () {
    const { suggestedBars, suggestedRestaurants } = this.props;

    return (
      <section className="suggestedPlans">
        <div className="plansContainer">
          <h3>Suggested Plan</h3>
          { suggestedBars.length && suggestedRestaurants.length ? 
            this.getPlans() : null }
        </div>
      </section>
    );
  }
}

Plans.propTypes = {
  suggestedRestaurants: PropTypes.array,
  suggestedBars: PropTypes.array,
  selectedEvent: PropTypes.object
};

Stop.propTypes = {
  rating: PropTypes.number,
  selectedEvent: PropTypes.object
};

export const mapStateToProps = (state) => ({
  suggestedRestaurants: state.suggestedRestaurants,
  suggestedBars: state.suggestedBars,
  selectedEvent: state.selectedEvent,
  user: state.user
});

export default withRouter(connect(mapStateToProps)(Plans));