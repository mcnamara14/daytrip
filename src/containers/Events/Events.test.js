import React from 'react';
import ReactDOM from 'react-dom';
import { Events } from './Events';
import { shallow } from 'enzyme';
import { mockDirtyRecentEvents, mockCleanRestaurantAndBar } from '../../mockData';
import SelectedEvent from '../SelectedEvent/SelectedEvent';
import Plans from '../Plans/Plans';

describe('Events', () => {
  let wrapper;
  let mockEvents;
  let mockSuggestedRestaurants;
  let mockSuggestedBars;
  let mockSelectedEvent;

  beforeEach(() => {
    mockEvents = mockDirtyRecentEvents.events;
    mockSuggestedRestaurants = mockCleanRestaurantAndBar;
    mockSuggestedBars = mockCleanRestaurantAndBar;
    mockSelectedEvent = mockDirtyRecentEvents.events
    
    wrapper = 
      shallow(<Events 
        events={mockEvents} 
        suggestedRestaurants={mockSuggestedRestaurants}
        suggestedBars={mockSuggestedBars}
        selectedEvent={mockSelectedEvent}
      />);
  })
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should render a SelectedEvent component when an event is selected', () => {
    expect(wrapper.find(SelectedEvent).length).toEqual(1);
  })

  it('should recent recent events when no SelectedEvent is present', () => {
    mockSelectedEvent = '';
    wrapper = 
    shallow(<Events 
      events={mockEvents} 
      suggestedRestaurants={mockSuggestedRestaurants}
      suggestedBars={mockSuggestedBars}
      selectedEvent={mockSelectedEvent}
    />);

    expect(wrapper.find('.recentEvents').length).toEqual(1);
  })

  it('should render a Plans container when an there are suggested bars and restaurants', () => {
    expect(wrapper.find(Plans).length).toEqual(1);
  })

  it('should not render a Plans container when an there are nop suggested bars and restaurants', () => {
    mockSuggestedBars = [];
    wrapper = 
    shallow(<Events 
      events={mockEvents} 
      suggestedRestaurants={mockSuggestedRestaurants}
      suggestedBars={mockSuggestedBars}
      selectedEvent={mockSelectedEvent}
    />);

    expect(wrapper.find(Plans).length).toEqual(0);
  })
});
