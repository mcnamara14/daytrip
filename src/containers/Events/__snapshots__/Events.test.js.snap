// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Events should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Events
    events={
        Array [
            Object {
              "date": "2018-05-30 2018-05-27T17:13:38-06:00",
              "id": "Zrt9dk32l",
              "image": "https://s1.ticketm.allswifty.jpg",
              "latitude": "39.735001",
              "longitude": "-105.020401",
              "price": "$300+",
              "title": "T Swift",
              "venue": "Boulder Theater",
            },
          ]
    }
    selectedEvent={
        Object {
            "date": "2018-05-30 2018-05-27T17:13:38-06:00",
            "id": "Zrt9dk32l",
            "image": "https://s1.ticketm.allswifty.jpg",
            "latitude": "39.735001",
            "longitude": "-105.020401",
            "price": "$300+",
            "title": "T Swift",
            "venue": "Boulder Theater",
          }
    }
    suggestedBars={
        Array [
            Object {
              "address": "1420 E 18th Ave Denver, CO 80218",
              "image_url": "https://s3-media3.fl.yelpcdn.com/tg/o.jpg",
              "name": "Beebs",
              "price": "$$",
              "rating": 4.5,
              "review_count": 450,
            },
          ]
    }
    suggestedRestaurants={
        Array [
            Object {
              "address": "1420 E 18th Ave Denver, CO 80218",
              "image_url": "https://s3-media3.fl.yelpcdn.com/tg/o.jpg",
              "name": "Beebs",
              "price": "$$",
              "rating": 4.5,
              "review_count": 450,
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown />,
        <div
          className="eventsStopsSelection"
>
          <Connect(EventsSearch) />
          <Connect(StopsSelection) />
</div>,
        <Connect(SelectedEvent) />,
        <Connect(Plans) />,
      ],
      "className": "eventsContainer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Connect(EventsSearch) />,
            <Connect(StopsSelection) />,
          ],
          "className": "eventsStopsSelection",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown />,
          <div
            className="eventsStopsSelection"
>
            <Connect(EventsSearch) />
            <Connect(StopsSelection) />
</div>,
          <Connect(SelectedEvent) />,
          <Connect(Plans) />,
        ],
        "className": "eventsContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Connect(EventsSearch) />,
              <Connect(StopsSelection) />,
            ],
            "className": "eventsStopsSelection",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
